fsharp-backend/src/BackendOnlyStdLib/LibDarkInternal.fs

```diff
@@ -765,6 +765,112 @@ human-readable data."
+      description = "Records an in-progress static asset deployment"
+      fn =
+        internalFn (function
+          | _, [ DStr username; DUuid canvasID ] ->
+            uply {
+              match! Account.getUser (UserName.create username) with
+              | None ->
+                Exception.raiseInternal $"User not found" [ "username", username ]
+                return DResult(Error(DStr "User not found"))
+              | Some user ->
+                let! canvasMeta = Canvas.getMetaFromID canvasID
+                let! deploy =
+                  StaticAssets.startStaticAssetDeploy user canvasID canvasMeta.name
+                return
+                  DObj(
+                    Map [ "deployHash", DStr deploy.deployHash
+                          "url", DStr deploy.url
+                          "status", DStr(string deploy.status)
+                          "lastUpdate",
+                          DDate(DDateTime.fromInstant deploy.lastUpdate) ]
+                  )
+                  |> Ok
+                  |> DResult
+            }
+          | _ -> incorrectArgs ())
+      sqlSpec = NotQueryable
+      previewable = Impure
+      deprecated = NotDeprecated }
+
+
+    // TODO should this take username as a parameter?
+    // should user B be able to finish user A's upload?
+    { name = fn "DarkInternal" "finishStaticAssetDeploy" 0
+      parameters =
+        [ Param.make "canvasID" TUuid ""; Param.make "deployHash" TStr "" ]
+      returnType =
+        TResult(
+          TRecord [ "deployHash", TStr
+                    "url", TStr
+                    "status", TStr
+                    "lastUpdate", TDate ],
+          TStr
+        )
+      description = "Marks an in-progress static asset deployment as finished"
+      fn =
+        internalFn (function
+          | _, [ DUuid canvasID; DStr deployHash ] ->
+            uply {
+              let! canvasMeta = Canvas.getMetaFromID canvasID
+              let! deploy =
+                StaticAssets.finishStaticAssetDeploy
+                  canvasID
+                  canvasMeta.name
+                  deployHash
+              return
+                DObj(
+                  Map [ "deployHash", DStr deploy.deployHash
+                        "url", DStr deploy.url
+                        "status", DStr(string deploy.status)
+                        "lastUpdate", DDate(DDateTime.fromInstant deploy.lastUpdate) ]
+                )
+                |> Ok
+                |> DResult
+            }
+          | _ -> incorrectArgs ())
+      sqlSpec = NotQueryable
+      previewable = Impure
+      deprecated = NotDeprecated }
+
+
+    { name = fn "DarkInternal" "deleteStaticAssetDeploy" 0
+      parameters =
+        [ Param.make "username" TStr ""
+          Param.make "canvasID" TUuid ""
+          Param.make "deployHash" TStr "" ]
+      returnType =
+        // TODO what should the return type be here? I basically want an empty result.
+        TResult(TNull, TStr)
```

Comment:
What's a good return type for `deleteStaticAssetDeploy`? Intuitively I want it to be a `Result` of some sort, but wouldn't expect any "data" to come back. `Result<unit, string>` is what I'd normally reach for, but since we don't have `unit`, `null` is the best idea I have.